pixels/behaviours/base.py

```diff
@@ -412,21 +412,53 @@ def process_lfp(self):
             print("> Mapping LFP data")
             data = ioutils.read_bin(data_file, num_chans)
 
+            output = self.processed / recording['lfp_processed']
+            if output.exists():
+                continue
+
+            print("> Performing median subtraction across channels for each timepoint")
+            subtracted = signal.median_subtraction(data, axis=1)
+
             print(f"> Downsampling to {self.sample_rate} Hz")
-            data = signal.resample(data, orig_rate, self.sample_rate)
+            downsampled = signal.resample(subtracted, orig_rate, self.sample_rate, False)
+            sync_chan = downsampled[:, -1]
+            downsampled = downsampled[:, :-1]
 
+            """
             if self._lag[rec_num] is None:
                 self.sync_data(rec_num, sync_channel=data[:, -1])
             lag_start, lag_end = self._lag[rec_num]
+            """
+            sd = self.processed / recording['lfp_sd']
+            if sd.exists():
+                continue
 
-            output = self.processed / recording['lfp_processed']
-            print(f"> Saving data to {output}")
+            SDs = []
+            for i in range(downsampled.shape[1]):
+                SDs.append(np.std(downsampled[:, i]))
+            results = dict(
+                median=np.median(SDs),
+                SDs=SDs,
+            )
+            print(f"> Saving standard deviation (and their median) of each channel")
+            with open(sd, 'w') as fd:
+                json.dump(results, fd)
+
+            """
             if lag_end < 0:
                 data = data[:lag_end]
             if lag_start < 0:
                 data = data[- lag_start:]
```

Comment:
How come you've removed trimming the lag at the ends?

Reply:
oh right i commented it out for my own usage, should not have pushed it, will fix it